model/connect/MySQLiConnector.php

```diff
@@ -235,18 +238,16 @@ public function preparedQuery($sql, $parameters, $errorLevel = E_USER_ERROR) {
 			$statement->execute();
 			return $statement;
 		});
-
-		// check result
-		$this->setLastStatement($lastStatement);
+		
 		if (!$lastStatement || $lastStatement->error) {
 			$values = $this->parameterValues($parameters);
 			$this->databaseError($this->getLastError(), $errorLevel, $sql, $values);
 			return null;
 		}
 
-		// May not return result for non-select statements
-		if($result = $lastStatement->get_result()) {
-			return new MySQLQuery($this, $result, $lastStatement);
+		// Non-select queries will have no result data
+		if($lastStatement && ($metaData = $lastStatement->result_metadata())) {
```

Comment:
So, are you saying that you use MySQL prepared statements even when there are no variables to pass in? 

What happens if the query doesn't throw an error but also doesn't provide meta-data? Maybe just throw an exception if that happens?